pkg/autodiscovery/autoconfig.go

```diff
@@ -142,6 +154,53 @@ func (ac *AutoConfig) checkTagFreshness(ctx context.Context) {
 	}
 }
 
+// checkBareServices looks for services that have no configs, and eventually
+// schedules an "empty" config for such services.
+func (ac *AutoConfig) checkBareServices(ctx context.Context) {
```

Comment:
Beware that this leaks memory as is, since it doesn't unschedule services as they go away. Nor does it handle going from bare to non-bare and the other way around (see my comment above).